### PR TITLE
feat(rpc): Add RPC endpoint for Fungible Tokens

### LINKED ISSUE #164: Create RPC endpoint to retrieve withdrawal info for fungible tokens
I noticed that we only have implemented the withdrawal information RPC endpoint for NFTs (https://github.com/hirosystems/stacks-subnets/pull/131) and for STX (https://github.com/hirosystems/stacks-subnets/pull/116). 

We need to implement a similar endpoint for fungible tokens.

### PR DESCRIPTION
**Pull request by fpbgg**
### Applicable issues
- fixes #164

### Checklist
- [x] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

**Comment by obycode** [comment]
How is this coming along @fpbgg? Is it ready for review? Can I help get this PR over the finish line tomorrow?

**Comment by fpbgg** [comment]
Thanks @obycode I was trying to figure out all the warnings and then realized they were caused by one error I had introduced in the unit test code.

**Comment by obycode** [comment]
> trying to figure out all the warnings

There are a bunch of warnings from merging the stacks-node changes over that need to be cleaned up. Don't worry about those for now. We can tackle those after this round of tasks.

File: src/net/http.rs [hunk 1]
```diff
@@ -1859,6 +1859,50 @@ impl HttpRequestType {
...
+    fn parse_get_ft_withdrawal<R: Read>(
+        _protocol: &mut StacksHttp,
+        preamble: &HttpRequestPreamble,
+        captures: &Captures,
+        _query: Option<&str>,
+        _fd: &mut R,
+    ) -> Result<HttpRequestType, net_error> {
+        if preamble.get_content_length() != 0 {
+            return Err(net_error::DeserializeError(
+                "Invalid Http request: expected 0-length body for GetAccount".to_string(),
```
Comment (obycode) [inline src/net/http.rs:1871]:
```suggestion
                "Invalid Http request: expected 0-length body for GetNFTWithdrawal".to_string(),
```

File: src/net/http.rs [hunk 2]
```diff
@@ -1859,6 +1859,50 @@ impl HttpRequestType {
...
+
+        let withdrawal_id = u32::from_str(&captures["withdrawal_id"])
+            .map_err(|_e| net_error::DeserializeError("Failed to parse block height".into()))?;
+        let contract_addr =
+            StacksAddress::from_string(&captures["contract_address"]).ok_or_else(|| {
+                net_error::DeserializeError("Failed to parse contract address".into())
+            })?;
+        let contract_name = ContractName::try_from(captures["contract_name"].to_string())
+            .map_err(|_e| net_error::DeserializeError("Failed to parse contract name".into()))?;
+        let id = u128::from_str(&captures["id"])
```
Comment (obycode) [inline src/net/http.rs:1890]: Let's make this `amount` for FTs instead of `id`.

File: src/net/http.rs (hunk 1 above)
Comment (obycode) [reply src/net/http.rs:1871]: I just noticed `parse_get_nft_withdrawal` has this wrong too. Can you update that as well while you're in here?

**Review by obycode — CHANGES_REQUESTED** [review]
Thanks @fpbgg. This is looking good. The main change I'd like to request is to use `amount` in place of `id` everywhere that is referencing FTs. I also need to review your changes in the tests farther.

**Comment by obycode** [comment]
#218 has the remaining fixes for this.

**Pull request closed without merging**